pennylane/vqe/vqe.py

```diff
@@ -90,36 +92,41 @@ class Hamiltonian:
         Note that this issue also arises when calling the ``simplify()`` method.
     """
 
-    def __init__(self, coeffs, observables, simplify=False):
+    num_wires = qml.operation.AnyWires
+    num_params = 1
+    par_domain = "A"
+
+    def __init__(self, coeffs, observables, simplify=False, id=None):
 
-        if len(coeffs) != len(observables):
+        if qml.math.shape(coeffs)[0] != len(observables):
             raise ValueError(
                 "Could not create valid Hamiltonian; "
                 "number of coefficients and operators does not match."
             )
 
-        if any(np.imag(coeffs) != 0):
-            raise ValueError(
-                "Could not create valid Hamiltonian; " "coefficients are not real-valued."
-            )
-
         for obs in observables:
             if not isinstance(obs, Observable):
                 raise ValueError(
                     "Could not create circuits. Some or all observables are not valid."
                 )
 
-        self._coeffs = list(coeffs)
+        self._coeffs = coeffs
         self._ops = list(observables)
+        self._wires = qml.wires.Wires.all_wires([op.wires for op in self.ops], sort=True)
 
         self.data = []
         self.return_type = None
 
+        self._grouped_coeffs = None
+        self._grouped_obs = None
+
```

Comment:
These get initialised when `Hamiltonian.group()` is called, and can be used by devices or QNodes to extract grouping information.